### PR TITLE
[rush-lib] Add publish command line arg tag to git tag

### PR DESCRIPTION
**Pull request by ethan-burrell**
Linked issue: https://github.com/microsoft/rushstack/issues/3222

These changes address the bug here (in part 4): https://github.com/microsoft/rushstack/issues/2536

## Summary

This PR fixes how git tags are added after a `prerelease` is published.

If the publish command was: `rush publish --apply --publish --tag beta --prerelease-name beta`

The old git tag would be:
`projectName_projectVersion`

With this change the tag will be:
`projectName_projectVersion-beta`

## Details

This is a problem because it can cause conflicting git tags, which could cause pipelines to fail before they finish building.

## How it was tested

- [ ] Unit tests
- [ ] Integration tests

### CODE REVIEW

File: apps/rush-lib/src/logic/test/PublishGit.test.ts
Comment (iclanton) [inline apps/rush-lib/src/logic/test/PublishGit.test.ts:2]:
```suggestion
import * as path from 'path';

import { RushConfiguration } from "../../api/RushConfiguration";
```

File: common/changes/@microsoft/rush/fix-git-tagging_2022-02-09-23-28.json
Comment (iclanton) [inline common/changes/@microsoft/rush/fix-git-tagging_2022-02-09-23-28.json:5]:
```suggestion
      "comment": "Fix an issue where the git tag would not include the prerelease portion of the version, which will be overwritten if a non-prerelease of the same version is published.",
```

File: apps/rush-lib/src/logic/test/PublishGit.test.ts
Comment (iclanton) [inline apps/rush-lib/src/logic/test/PublishGit.test.ts:47]:
```suggestion
    const rushFilename: string = path.resolve(__dirname, '../../api/test/repo/rush-npm.json');
```

File: apps/rush-lib/src/logic/test/PublishGit.test.ts
Comment (iclanton) [inline apps/rush-lib/src/logic/test/PublishGit.test.ts:45]:
```suggestion
    const execCommand = jest.spyOn(PublishUtilities, 'execCommand').mockImplementation(() => { /* no-op */ });
```

File: apps/rush-lib/src/logic/test/PublishGit.test.ts
Comment (iclanton) [inline apps/rush-lib/src/logic/test/PublishGit.test.ts:16]:
```suggestion
    const execCommand = jest.spyOn(PublishUtilities, 'execCommand').mockImplementation(() => { /* no-op */ });
```

File: apps/rush-lib/src/logic/test/PublishGit.test.ts
Comment (iclanton) [inline apps/rush-lib/src/logic/test/PublishGit.test.ts:18]:
```suggestion
    const rushFilename: string = path.resolve(__dirname, '../../api/test/repo/rush-npm.json');
```

File: apps/rush-lib/src/logic/PublishGit.ts
Comment (iclanton) [inline apps/rush-lib/src/logic/PublishGit.ts:107]: Instead of using the NPM tag, wouldn't it be better to use the real version? Like if the npm tag is "dev", but you publish versions `1.0.0-dev.0` and `1.0.0-dev1`, both with the `dev` tag, they'll both end up with the same Git tag.

Comment (iclanton) [reply apps/rush-lib/src/logic/PublishGit.ts]: That parameter does have a confusing name. Yeah, it should be the prerelease name.

**Comment by ethan-burrell** [comment]
> Looks good overall. Can you sign the CLA?

@iclanton just got approval from my company for CLA, I'm ready to merge if you approve.

**Review by iclanton — APPROVED** [review]

**Pull request merged**
